src/controller/CHIPDeviceController.cpp

```diff
@@ -142,10 +142,13 @@ CHIP_ERROR DeviceController::Init(ControllerInitParams params)
     mOperationalCredentialsDelegate = params.operationalCredentialsDelegate;
 
     mFabricIndex = params.fabricIndex;
-    ReturnErrorOnFailure(ProcessControllerNOCChain(params));
-
-    mFabricInfo = params.systemState->Fabrics()->FindFabricWithIndex(mFabricIndex);
```

Comment:
This cannot be known in advance. This is basically trying to short-cut the invoker of chip-tool having knowledge of fabric IDs and roots. I think we should make it clear that:

1- chip-tool et al use a particular root of trust, implied across invocations: "a default root"
2- Fabric info should be looked-up by root + fabric ID
3- Root can be passed by the caller, since it should know it

Reply:
OK, that's easy enough.

Will do.

Reply:
Done exactly as above.  We pass the operational key pair and cert chain to the commissioner init, and it uses these to instantiate itself, looking up the fabric in the fabric table by root + fabric ID.